File: misc/sumseq.cpp
#include <iostream>
#include <stdlib.h>
#include <time.h>
#include <limits.h>
using namespace std;

int sumArray(int* arr, int length) {
	int total = 0;
	for(int i = 0; i < length; i++) {
		total += arr[i];
	}
	return total;
}

int main() {
	// Generate numbers
	const int ARRAY_LENGTH = 10;
	int numbers[ARRAY_LENGTH];
	srand(time(NULL));
	int temp;
	cout << "[ ";
	for(int i = 0; i < ARRAY_LENGTH; i++) {
		temp = rand() % 10;
		temp % 2 == 0? numbers[i] = temp : numbers[i] = -1 * temp;
		cout << numbers[i];
		if(i < ARRAY_LENGTH - 1) {
			cout << ", ";
		}
	}
	cout << " ]\n";

	int first, last = 0;
	int startIndex, lastIndex = 0;
	int maxSum = INT_MIN;
	int currentSum = 0;

	for(first = 0; first < ARRAY_LENGTH; first++) {
		for(last = first + 1; last < ARRAY_LENGTH; last++) {
			currentSum = sumArray(numbers + first, last - first);
			if(currentSum > maxSum) {
				maxSum = currentSum;
				startIndex = first;
				lastIndex = last;
			}
		}
	}

	cout << "Largest sum is " << maxSum << " from " << startIndex << " to " << lastIndex << ", ";
	for(int i = startIndex; i <= lastIndex; i++) {
		cout << numbers[i];
		if (i < lastIndex) {
			cout << ", ";
		}
		else {
			cout << endl;
		}
	}
}

File: misc/sumseq.py
import sys
from random import randint

def largestSequence(numbers):
	# numbers holds integers. This method finds the largest sum obtained by summing
	# consecutive elements in numbers and details the sum and the sequence that yielded it. 
	print numbers
	startIndex = 0
	endIndex = 0
	maxSum = -sys.maxint - 1 # Smallest possible integer in Python 2.x, 3.x seems to be different?
	listLength = len(numbers)
	currentSum = 0;

	for first in range(0, listLength):
		for last in range(first, listLength):
			currentSum = sum(numbers[first:last + 1])
			if currentSum > maxSum:
				maxSum = currentSum
				startIndex = first
				endIndex = last

	print 'Largest sum is ' + str(maxSum) + ' from ' + str(startIndex) + ' to ' + str(endIndex) + ', ' + str(numbers[startIndex:endIndex + 1])

if __name__ == '__main__':
	numbers = []
	# Generate random integers and pass them to largestSequence()
	for i in range(0, 10):
		numbers.append(randint(-10, 10))

	largestSequence(numbers)